plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p5edges/PolylineEdgeRouter.java

```diff
@@ -323,6 +323,23 @@ private void processNode(final LNode node, final double layerLeftXPos, final dou
         
         for (LPort port : node.getPorts()) {
             KVector absolutePortAnchor = port.getAbsoluteAnchor();
+            
+            if (node.getType() == NodeType.NORTH_SOUTH_PORT) {
+                // North/south ports require special handling (see also #515):
+                // The dummy node that represents the north/south port will be placed at the very left x-coordinate
+                // of the layer currently processed. However, the actual x-coordinate of the port itself is very likely
+                // at a larger x-coordinate.
```

Comment:
Might also be placed at the layer's right border if there are only outgoing edges, right?

Reply:
Good point. I didn't check that. The fix should still be the same, I guess?
I'll add a test.

Reply:
Behavior as expected: dummy placed at the very right of the layer for an outgoing edge - fix is the same. 

Note that the issue does not occur for north/south ports with incoming _and_ outgoing edges when the dummy is placed somewhere inside the layer. I added a testfile for this case as well, however.

Reply:
Those dummy nodes are placed just like regular nodes: their position inside the layer depends on the number of connections to each side, which sort of "pull" the node close to this or that border of the layer. The fix will be the same in all three conditions.

Reply:
I was trying to say: If there is one incoming and one outgoing edge, there's no need for a fix, as it's working properly already :).

Reply:
That depends, doesn't it? Perhaps the port is placed in a way that it ends up at the layer's left boundary. Thus, no additional bend point might be required for edges from the left.

Reply:
That's why I added "when the dummy is placed somewhere inside the layer" to my sentence (in my first answer 🙄).

Anyway, anything preventing this from merging?

Reply:
Nope, I just wanted the comment to not confuse anyone who might be trying to debug any of this later on. 🙂